makefile

```diff
@@ -6,17 +6,28 @@ build_folder := $(shell mkdir -p build)
 bin_folder := $(shell mkdir -p bin)
 INCLUDECADMIUM=-I lib/cadmium-master/include
 
-all:
+
+all: build/main.o build/main_r.o build/main_s.o build/main_n.o
 	$(CC) -g -o bin/ABP build/main.o build/message.o 
 	$(CC) -g -o bin/RECEIVER_TEST build/main_r.o build/message.o
 	$(CC) -g -o bin/SENDER_TEST build/main_s.o build/message.o
 	$(CC) -g -o bin/SUBNET_TEST build/main_n.o build/message.o
 
-comp:
+comp: main message main_r main_s main_n
```

Comment:
You may consider re-arranging main_r, main_s, main_n as main_s, main_n, main_r. (i.e., sender  subnet  receiver). Same for all statement as well. 
blank lines in the make file can be deleted.

Reply:
I have updated the makefile as you mentioned and for blank lines we added as it makes it look more readable